compute/reservations/reservations_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/protobuf/proto"
+)
+
+func createTemplate(ctx context.Context, project, templateName string) error {
+	client, err := compute.NewInstanceTemplatesRESTClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	disk := &computepb.AttachedDisk{
+		AutoDelete: proto.Bool(true),
+		Boot:       proto.Bool(true),
+		InitializeParams: &computepb.AttachedDiskInitializeParams{
+			SourceImage: proto.String("projects/debian-cloud/global/images/family/debian-12"),
+			DiskSizeGb:  proto.Int64(25),
+			DiskType:    proto.String("pd-balanced"),
+		},
+	}
+	req := &computepb.InsertInstanceTemplateRequest{
+		Project: project,
+		InstanceTemplateResource: &computepb.InstanceTemplate{
+			Name: proto.String(templateName),
+			Properties: &computepb.InstanceProperties{
+				MachineType: proto.String("n1-standard-4"),
+				Disks:       []*computepb.AttachedDisk{disk},
+				NetworkInterfaces: []*computepb.NetworkInterface{{
+					Name: proto.String("global/networks/default"),
+				}},
+			},
+		},
+	}
+	op, err := client.Insert(ctx, req)
+	if err != nil {
+		return err
+	}
+	return op.Wait(ctx)
+}
+
+func getTemplate(ctx context.Context, project, templateName string) (*computepb.InstanceTemplate, error) {
+	client, err := compute.NewInstanceTemplatesRESTClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &computepb.GetInstanceTemplateRequest{
+		Project:          project,
+		InstanceTemplate: templateName,
+	}
+	return client.Get(ctx, req)
+}
+
+func deleteTemplate(ctx context.Context, project, templateName string) error {
```

Comment:
issue: remove context object from method signature. Create a new context object in the function.